lib/resources/nginx_conf.rb

```diff
@@ -92,4 +101,92 @@ def resolve_references(data, rel_path)
       Hash[data.map { |k, v| [k, resolve_references(v, rel_path)] }]
     end
   end
+
+  class NginxConfHttp
+    attr_reader :entries
+    def initialize(params, parent)
+      @parent = parent
+      @entries = params.map { |x| NginxConfHttpEntry.new(x, parent) }
+    end
+
+    def servers
+      @entries.map(&:servers).flatten
+    end
+
+    def locations
+      servers.map(&:locations).flatten
+    end
+
+    def to_s
+      @parent.to_s + ', http entries'
+    end
+    alias inspect to_s
+  end
+
+  class NginxConfHttpEntry
+    attr_reader :params, :parent
+    def initialize(params, parent)
+      @params = params
+      @parent = parent
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add(:servers, field: 'server')
+          .connect(self, :server_table)
+
+    def locations
+      servers.map(&:locations).flatten
+    end
+
+    def to_s
+      @parent.to_s + ', http entry'
+    end
+    alias inspect to_s
+
+    private
+
+    def server_table
+      @server_table ||= (params['server'] || []).map { |x| { 'server' => NginxConfServer.new(x, self) } }
+    end
+  end
+
+  class NginxConfServer
+    attr_reader :params, :parent
+    def initialize(params, parent)
+      @parent = parent
+      @params = params
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add(:locations, field: 'location')
+          .connect(self, :location_table)
+
+    def to_s
+      # go two levels up: 1. to the http entry and 2. to the root nginx conf
+      @parent.parent.to_s + ', server entry'
```

Comment:
I think we can be even more transparent with users, especially when they're debugging / exploring in InSpec shell. Otherwise you get output like this:

```
inspec> a.http.servers.entries
=> [nginx_conf /Users/aleff/projects/inspec/test/unit/mock/files/nginx.conf, server entry,
 nginx_conf /Users/aleff/projects/inspec/test/unit/mock/files/nginx.conf, server entry]
```

Can we do something like this instead?

```
def to_s
  server_name = ''
  unless params['server_name'].flatten.nil?
    server_name = params['server_name'].flatten.first
    server_name += ":#{params['listen'].flatten} unless params['listen'].flatten.nil?
  end
  @parent.parent.to_s + ", #{server_name} server entry"
end
```

^^^ not perfectly tested, but hopefully you get the gist of what I'm trying to describe.

Reply:
Yeah, totally! I wanted to leave it out of this PR to figure out pretty-printing for all 3 new nginx portions in a follow-up PR. Are you ok if we issue this in a follow-up?

Reply:
Added both of these to this PR, thank you for the discussion!